src/libstd/sys/unix/fd.rs

```diff
@@ -51,12 +51,20 @@ impl FileDesc {
         Ok(ret as usize)
     }
 
+    #[cfg(not(target_env = "newlib"))]
     pub fn set_cloexec(&self) {
         unsafe {
             let ret = c::ioctl(self.fd, c::FIOCLEX);
             debug_assert_eq!(ret, 0);
         }
     }
+    #[cfg(target_env = "newlib")]
+    pub fn set_cloexec(&self) {
+        unsafe {
+            let ret = c::fnctl(self.fd, c::F_SETFD, c::FD_CLOEXEC);
```

Comment:
Hm. There *are* other flags (`O_NONBLOCK`)... So facepalm on that bit. Anyway, fixed.